cmd/attest.go

```diff
@@ -69,12 +76,23 @@ hardware and guarantees a fresh quote.
 
 		attestOpts := client.AttestOpts{}
 		attestOpts.Nonce = nonce
-		if len(teeNonce) != 0 {
-			attestOpts.TEENonce = teeNonce
-			attestOpts.TEEDevice, err = client.CreateSevSnpDevice()
-			if err != nil {
-				return fmt.Errorf("failed to collect TEE attestation report: %v", err)
+		if len(teeTechnology) != 0 {
+			// Add logic to open other hardware devices when required.
+			switch teeTechnology {
+			case SevSnp:
+				attestOpts.TEEDevice, err = client.CreateSevSnpDevice()
+				if err != nil {
+					return fmt.Errorf("failed to open %s device: %v", SevSnp, err)
+				}
+			default:
+				// Change the return statement when more devices are added
+				return fmt.Errorf("tee_technology should be empty or %s", SevSnp)
 			}
+			if len(teeNonce) != 0 {
+				attestOpts.TEENonce = teeNonce
+			}
+		} else if len(teeNonce) != 0 {
+			return fmt.Errorf("use of --teenonce requires specifying TEE hardware type with --tee_technology")
```

Comment:
Could we change "--teenonce" to be "--tee-nonce" for the naming convention? (I understand this flag is created in this PR)

Reply:
I have changed it to `--tee-nonce`